src/mpv.rs

```diff
@@ -96,19 +99,94 @@ unsafe impl GetData for i64 {
     }
 }
 
-unsafe impl SetData for i64 {
-    fn get_format() -> Format {
-        Format::Int64
+#[derive(Debug)]
+pub enum MpvNode<'a> {
```

Comment:
The interaction of the `MpvStr` and `MpvNode` is a bit tricky here with regards to the drop behaviour. It would probably be nicer to switch them to `&str` and call `mpv_free_node_contents` in a `Drop` impl, because like this I'm not sure if all the memory that was originally allocated by libmpv would be dropped correctly.

Reply:
That makes sense. I completely overlooked `mpv_free_node_contents` heh.